python-avd/pyavd/_eos_designs/shared_utils/routing.py

```diff
@@ -29,11 +29,13 @@ def underlay_routing_protocol(self: SharedUtils) -> str:
     @cached_property
     def overlay_routing_protocol(self: SharedUtils) -> str:
         default_overlay_routing_protocol = self.node_type_key_data.default_overlay_routing_protocol
+        if self.is_wan_router and self.inputs.wan_use_evpn_node_settings_for_lan:
+            default_overlay_routing_protocol = "none"
         return (self.inputs.overlay_routing_protocol or default_overlay_routing_protocol).lower()
 
     @cached_property
     def overlay_address_families(self: SharedUtils) -> list[str]:
-        if self.overlay_routing_protocol in ["ebgp", "ibgp"]:
+        if self.overlay_routing_protocol in ["ebgp", "ibgp"] or self.is_wan_router:
```

Comment:
Do we need this? Would it not be clearer if we add the wan_router handling in the places where this is checked today?

Reply:
so in filtered_tenants we are looping over these address families to see if we select a VRF. Which means we would need to check if the af of the vrf.address_families are in the overlay_address_families or if it is as WAN router AND the af is epvn

Reply:
I would prefer if we remove all ties to `overlay_address_families` from WAN, since it would be the only way the user could prevent a VRF from going to the LAN in a future state. (one VRF only being on WAN and local interfaces, but not extended to LAN)

Reply:
implemented